<?xml version="1.0"?>
<launch>
  
  <!--
    Mia Hand Hardware and Controller Arguments
    These arguments configure the Mia hand control parameters.
  -->
  <arg name="Mia_COM_" default="0" doc="Int number of the Mia hand COM port"/>
  <arg name="Mia_fs_" default="20" doc="Hz, frequency of the control loop of the Mia Hw interface"/>
  <arg name="robotNamespace" default="mia_hand_hw"/>
  <arg name="load_default_mia_model" default="false"/>
  <arg name="controller_thumb_fle" default="pos_vel" doc="Controller type for thumb flexion. Options: position, velocity, pos_vel"/>
  <arg name="controller_index_fle" default="pos_vel" doc="Controller type for index flexion. Options: position, velocity, pos_vel"/>
  <arg name="controller_mrl_fle" default="pos_vel" doc="Controller type for mrl flexion. Options: position, velocity, pos_vel"/>
  <arg name="Rviz_on" default="false" doc="If true, start Rviz to visualize Mia hand joint state"/>

  <!-- Include Mia Hand Hardware launch file -->
  <include file="$(find mia_hand_bringup)/launch/mia_hand_hw.launch">
    <arg name="Mia_COM_" value="$(arg Mia_COM_)"/>
    <arg name="Mia_fs_" value="$(arg Mia_fs_)"/>
    <arg name="robotNamespace" value="$(arg robotNamespace)"/>
    <arg name="load_default_mia_model" value="$(arg load_default_mia_model)"/>
    <arg name="controller_thumb_fle" value="$(arg controller_thumb_fle)"/>
    <arg name="controller_index_fle" value="$(arg controller_index_fle)"/>
    <arg name="controller_mrl_fle" value="$(arg controller_mrl_fle)"/>
    <arg name="Rviz_on" value="$(arg Rviz_on)"/>
  </include>

  <!-- user args -->
  <arg name="port" default="/dev/ttyACM0" doc="Port used by the Myo dongle. Check it with 'ls /dev/' and look for ttyACM0 for instance. You need to provide the full path." />

  <!-- myo ros node -->
  <node pkg="ros_myo" name="myo_raw" type="myo-rawNode.py" args="$(arg port)" output="screen"/>

  <!-- Launch your custom EMG controller node INSIDE mia_hand_hw namespace -->
  <group ns="mia_hand_hw">
    <node pkg="hand_band_control"
          type="hand_band_controller.py"
          name="hand_band_controller"
          output="screen" />
  </group>
</launch>